Dshop/apps/products_catalogue/tests/test_api_persistent_cart.py

```diff
@@ -0,0 +1,163 @@
+from django.forms import ValidationError
+from django.urls import reverse
+from django.contrib.auth.models import User
+import pytest
+from rest_framework.test import APIClient
+from rest_framework import status
+
+
+def counts_to_price_quantity(products, quantities):
+    assert len(products) == len(quantities)
+    quant_map = enumerate(quantities)
+    total_quantity = sum(quantities)
+    total_price = sum(products[idx].price * quantity for idx, quantity in quant_map)
+    return total_quantity, total_price
+
+
+@pytest.mark.django_db
+def test_get_cart_detail():
+    client = APIClient()
+    url = reverse("api_cart_detail")
+    user = User.objects.create_user(username='testuser', password='testpassword')
+    client.force_authenticate(user)
+    response = client.get(url, {}, content_type="application/json")
+    assert response.status_code == status.HTTP_200_OK
+    assert "total" in response.data
+    assert "count" in response.data
+    assert "items" in response.data
+
+
+@pytest.mark.django_db
+def test_add(tv_product):
+    client = APIClient()
+    product = tv_product
+    add_url = reverse("api_add_to_cart", args=[product.id, 10])
+    user = User.objects.create_user(username='testuser', password='testpassword')
+    client.force_authenticate(user)
+    response = client.post(add_url, {}, content_type="application/json")
+    data = response.data
+    first_item = data.get("items")[0]
```

Comment:
In tests, IMO, it's better to not use .get. W/o it, it makes test fail cleanly with an exception, here if the key is missing you will not get KeyError but AssertionError. 
Please change. In all places.

Reply:
done